### PR TITLE
Reorganize project files

### PR DESCRIPTION
**Pull request by matteblair**
Working towards the ideas in: #1212 

 - 3rd party dependencies located in sensible places based on their usage
 - Public headers for core library located in a single folder
 - Platform code in separate folders in a top-level platforms folder

Coming later:

 - Split main README contents into separate files for each platform, link from main README
 - Replace some submodules with plain source trees

### CODE REVIEW

**Comment by lygstate** [comment]
Waiting for finish.

**Comment by matteblair** [comment]
Nearly done! Android builds are succeeding in the Release configuration, but failing in Debug with:
```
FAILED: : && /Users/matt/Development/android-sdk/ndk-bundle/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang++  -target armv7-none-linux-androideabi -gcc-toolchain /Users/matt/Development/android-sdk/ndk-bundle/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64 --sysroot=/Users/matt/Development/android-sdk/ndk-bundle/platforms/android-15/arch-arm -fPIC -g -DANDROID -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -march=armv7-a -mfloat-abi=softfp -mfpu=vfpv3-d16 -fno-integrated-as -mthumb -Wa,--noexecstack -Wformat -Werror=format-security -fno-exceptions -fno-rtti -std=c++11 -std=c++14 -pedantic -fPIC -fexceptions -frtti -g -O0 -fno-limit-debug-info  -Wl,--build-id -Wl,--warn-shared-textrel -Wl,--fatal-warnings -Wl,--fix-cortex-a8 -Wl,--exclude-libs,libunwind.a -Wl,--no-undefined -Wl,-z,noexecstack -Qunused-arguments -Wl,-z,relro -Wl,-z,now -shared -Wl,-soname,libtangram.so -o ../../../../build/intermediates/cmake/slim/debug/obj/armeabi-v7a/libtangram.so CMakeFiles/tangram.dir/platforms/common/platform_gl.cpp.o CMakeFiles/tangram.dir/platforms/android/tangram/src/main/cpp/jniExports.cpp.o CMakeFiles/tangram.dir/platforms/android/tangram/src/main/cpp/platform_android.cpp.o  core/libcore.a -lGLESv2 -llog -lz -latomic -landroid core/deps/libduktape.a core/deps/libcss-color-parser-cpp.a core/deps/libgeojson-vt-cpp.a core/deps/yaml-cpp/libyaml-cpp.a core/deps/alfons/src/libalfons.a core/deps/alfons/src/liblinebreak.a core/deps/harfbuzz-icu-freetype/libharfbuzz.a core/deps/harfbuzz-icu-freetype/freetype-2.6/libfreetyped.a core/deps/harfbuzz-icu-freetype/libicucommon.a core/deps/SQLiteCpp/libSQLiteCpp.a core/deps/SQLiteCpp/sqlite3/libsqlite3.a -ldl -lm "/Users/matt/Development/android-sdk/ndk-bundle/sources/cxx-stl/llvm-libc++/libs/armeabi-v7a/libunwind.a" "/Users/matt/Development/android-sdk/ndk-bundle/sources/cxx-stl/llvm-libc++/libs/armeabi-v7a/libc++_shared.so" && :
  /Users/matt/Development/tangram-es/core/deps/harfbuzz-icu-freetype/freetype-2.6/src/gzip/inflate.c:136: error: undefined reference to 'z_verbose'
  /Users/matt/Development/tangram-es/core/deps/harfbuzz-icu-freetype/freetype-2.6/src/gzip/inflate.c:272: error: undefined reference to 'z_verbose'
  /Users/matt/Development/tangram-es/core/deps/harfbuzz-icu-freetype/freetype-2.6/src/gzip/inflate.c:272: error: undefined reference to 'z_verbose'
  /Users/matt/Development/tangram-es/core/deps/harfbuzz-icu-freetype/freetype-2.6/src/gzip/inflate.c:79: error: undefined reference to 'z_verbose'
  /Users/matt/Development/tangram-es/core/deps/harfbuzz-icu-freetype/freetype-2.6/src/gzip/infcodes.c:219: error: undefined reference to 'z_error'
  clang++: error: linker command failed with exit code 1 (use -v to see invocation)
  ninja: build stopped: subcommand failed.
```

**Comment by matteblair** [comment]
Fixed the Android build issue - there is a bit of a bug in Freetype 2 on Android that Hannes added a workaround for: https://github.com/tangrams/harfbuzz-icu-freetype/commit/43cdfca1183e4505f3238ed99db94f69b74e9ec0 but our submodule had not been updated to include that commit. I've updated the submodule here and that error is now resolved.

File: .gitmodules [hunk 1]
```diff
@@ -1,39 +1,39 @@
 [submodule "core/include/glm"]
-	path = core/include/glm
+	path = core/deps/glm
```
Comment (karimnaaji) [inline .gitmodules]: Are we still planning to directly put the _cold_ submodules as files within the repo?

Comment (matteblair) [inline .gitmodules]: Yes I'd like to do that in a subsequent PR

File: .gitmodules [hunk 2]
```diff
@@ -1,39 +1,39 @@
...
 	url = https://github.com/g-truc/glm.git
 [submodule "core/include/isect2d"]
-	path = core/include/isect2d
+	path = core/deps/isect2d
 	url = https://github.com/tangrams/isect2d.git
 [submodule "core/dependencies/csscolorparser"]
-	path = external/css-color-parser-cpp
+	path = core/deps/css-color-parser-cpp
```
Comment (matteblair) [inline .gitmodules]: Yes yes that is coming next

**Comment by karimnaaji** [comment]
Pushed an update to fix CI for docs 76a1fff.

**Comment by matteblair** [comment]
@hjanetzek Yeah, GLFW might be in a misleading location right now. I am hoping that we can combine the linux and osx platform folders into a glfw platform folder and then we can put it into `platforms/glfw/deps/`.

**Comment by tallytalwar** [comment]
Maybe `platform/desktop` instead of `platform/glfw`.

**Pull request merged**